Make requested changes from FoG PR

urwrstkn8mare opened the pull request:
I've tried to quickly do some of the essential requested changes in [this PR](https://github.com/FriendsOfGalaxy/galaxy-integration-minecraft/pull/8). Some changes I haven't implemented I've put as [TODO](https://github.com/urwrstkn8mare/galaxy-minecraft-integration#todo)s in the [README.md](https://github.com/TouwaStar/Galaxy_Plugin_Minecraft/pull/10/files#diff-04c6e90faac2675aa89e2176d2eec7d8) but I don't think they are necessary to merge with [FriendsOfGalaxy](https://github.com/FriendsOfGalaxy)'s [fork](https://github.com/FriendsOfGalaxy/galaxy-integration-minecraft).

Summary of what I've done:

- Add `INSTALL_MULTIMC.md` and redirect to it to make it easier to install.
- Added some cool badges!!!! 
- Updated GIF. 
- More minor changes to [`README.md`](https://github.com/TouwaStar/Galaxy_Plugin_Minecraft/pull/10/files#diff-04c6e90faac2675aa89e2176d2eec7d8) including adding TODOs.
- Simplify `launch()` function.
- Move `prepare_local_size_context()` logic to `get_local_size()`.
- Update Authentication UI.
- Move util type stuff to `utils` folder.
- use `with open()`.
- Return `GameTime`, not `None` in `get_game_time()`.
- Update `download()` to use separate thread.

urwrstkn8mare committed:
up ver

urwrstkn8mare committed:
add cool badges!

urwrstkn8mare committed:
update build.py to v1.0.0

urwrstkn8mare committed:
simply launch() function

should resolve https://github.com/FriendsOfGalaxy/galaxy-integration-minecraft/pull/8#discussion_r475073156

urwrstkn8mare committed:
move prepare_local_size_context() logic to get_local_size()

should resolve https://github.com/FriendsOfGalaxy/galaxy-integration-minecraft/pull/8#discussion_r475728742

urwrstkn8mare committed:
update authentication ui

should resolve https://github.com/FriendsOfGalaxy/galaxy-integration-minecraft/pull/8#discussion_r482564847 

should resolve https://github.com/FriendsOfGalaxy/galaxy-integration-minecraft/pull/8#discussion_r482597777

urwrstkn8mare committed:
add INSTALL_MULTIMC.md

urwrstkn8mare committed:
fix: remove asyncio.run() in pass_login_credentials

urwrstkn8mare committed:
make utils folder + move existing utils to it

urwrstkn8mare committed:
update utils imports

urwrstkn8mare committed:
refactor to use .get_tracking_games()

should resolve https://github.com/FriendsOfGalaxy/galaxy-integration-minecraft/pull/8#discussion_r482590313

urwrstkn8mare committed:
update double_click_effect() desc.

urwrstkn8mare committed:
expose exceptions + extended TimeTracker class via time_tracker.py

urwrstkn8mare committed:
fix: use 'with open()' instead

should resolve https://github.com/FriendsOfGalaxy/galaxy-integration-minecraft/pull/8#discussion_r482587147

urwrstkn8mare committed:
fix: return GameTime, not None

should resolve https://github.com/FriendsOfGalaxy/galaxy-integration-minecraft/pull/8#discussion_r482584257

urwrstkn8mare committed:
fix: update download() to use seperate thread

should resolve https://github.com/FriendsOfGalaxy/galaxy-integration-minecraft/pull/8#discussion_r482578788

urwrstkn8mare committed:
fix: remove imports that broke the integration

urwrstkn8mare committed:
fix: await download() to get launcher_path

urwrstkn8mare committed:
fix: update imports so it works

urwrstkn8mare committed:
change size of error window

urwrstkn8mare committed:
up ver

urwrstkn8mare committed:
update readme

remove covers + more

urwrstkn8mare committed:
Add TODO
update todo

upddate todo

urwrstkn8mare committed:
update install-multimc link to INSTALL_MULTIMC.md

urwrstkn8mare committed:
update update_href() function to add '&next=page2' automatically

urwrstkn8mare commented:
Sorry for all those commits and force-pushing. I had to rebase some stuff. You can go to the [commits](https://github.com/TouwaStar/Galaxy_Plugin_Minecraft/pull/10/commits) tab see the actual commits.

FriendsOfGalaxy commented on README.md:
```diff
@@ -1,18 +1,28 @@
 
+## Todo
+
+- [ ] Finish [INSTALL_MULTIMC.md](INSTALL_MULTIMC.md)
```
What should be finished here? Maybe prerequisites section with link to Java installation? Linking MultiMC Wiki page is good idea (not the best UX, but robust to any changes in external source)

FriendsOfGalaxy replied:
sth like:


```markdown
## Prerequisites:
MultiMC requires Java to be installed. Get installer here:
<java link I proposed last time>

## MultiMC Installation wiki:
<wiki link>

```

urwrstkn8mare replied:
My original plan was to link to the MultiMC Wiki page but also have some additional instructions detailing prerequisites (eg. installing Java), additional help on how to get the path to MultiMC to put into GOG Galaxy, and more. So far I haven't had time to do it but I've created the file for future development. I realise it may be better to just take the user directly to the [Getting Started](https://github.com/MultiMC/MultiMC5/wiki/Getting-Started) page until the [`INSTALL_MULTIMC.md`](https://github.com/TouwaStar/Galaxy_Plugin_Minecraft/pull/10/files#diff-869195911ccf6d7d5891f63309953cd7) is ready as they already have some light info about Java in there.

FriendsOfGalaxy commented on src/page/page3.html:
```diff
@@ -1,95 +1,111 @@
+            >
+              Ok
+            </button>
+            <a
+              class="forgot-password open"
+              href="https://github.com/TouwaStar/Galaxy_Plugin_Minecraft"
+            >
+              Integration Info
+            </a>
```
Do not reopen GUI if user click "Integration Info". If there is nothing else to do here, then just close it like with OK

FriendsOfGalaxy replied:
not obligatory

urwrstkn8mare replied:
good idea, thanks.

FriendsOfGalaxy commented on src/page/page2.html:
```diff
@@ -94,23 +102,45 @@ <h1 class="title">Set your MultiMC path</h1>
+            >
               Install MultiMC
             </a>
+            <a
+              class="forgot-password"
+              href="/finished?install_mc=true"
+              id="install-mc"
+              onclick="update_href(this.id)"
+            >
+              Install Minecraft
+            </a>
```
There is already a lot of info that overwhelm user. I prefer separate page for (Install Minecraft prompt & warning from above) that would be open if:
- user types any MultiMC path
- Vanila Minecraft is not installed (optional - nice to have)

FriendsOfGalaxy replied:
not obligatory

urwrstkn8mare replied:
Yeah, I see the issue with an overwhelming amount of info. It's definitely something that can be made better but unfortunately, I won't be able to implement it right now but instead, I'll add it to the todo.

FriendsOfGalaxy commented on src/utils/time_tracker.py:
```diff
@@ -0,0 +1,12 @@
+
+
+class TimeTracker(OGTimeTracker):
+    def get_tracking_games(self):
+        return [game_info.game_id for game_info in self._running_games_dict]
```
I got an error for minecraft dungeons when tried to play for the first time:
AttributeError: 'str' object has no attribute 'game_id'

urwrstkn8mare replied:
Yeah, I actually forgot to test that. Sorry, completely my fault. I'll fix that ASAP. Thanks for finding that issue.

urwrstkn8mare replied:
@FriendsOfGalaxy, I've just fixed it. I'd appreciate it if you could test it with Minecraft Dungeons or just Minecraft as I don't have the ability to test right now.

TouwaStar replied:
We might as well check it on the repo -> fog pull mr

urwrstkn8mare committed:
add/update todo

urwrstkn8mare committed:
make href links in page3 close window as well

should resolve https://github.com/TouwaStar/Galaxy_Plugin_Minecraft/pull/10#discussion_r486883453

urwrstkn8mare committed:
Moved requirements to subfolder

urwrstkn8mare committed:
Using tasks.py for dev instead now

urwrstkn8mare committed:
fix: should solve https://github.com/TouwaStar/Galaxy_Plugin_Minecraft/pull/10#discussion_r486900153

urwrstkn8mare committed:
add dev requirements

urwrstkn8mare committed:
updating build header to development for new tasks.py in README

urwrstkn8mare committed:
update install-multimc link to go to MultiMC's Getting Started

see https://github.com/TouwaStar/Galaxy_Plugin_Minecraft/pull/10#discussion_r486909188

TouwaStar commented on tasks.py:
```diff
@@ -0,0 +1,94 @@
+
+@task
+def pack(c):
+    output = "itch_" + MANIFEST["guid"]
```
probably not itch  :p

urwrstkn8mare replied:
Sorry about that! I've fixed and I also found an issue with the function where if the releases dir didn't exist, it would error. So I've fixed that as well.

See [0e393f3](https://github.com/TouwaStar/Galaxy_Plugin_Minecraft/pull/10/commits/0e393f3a4fb7b496e0444c06a8ef32530ccbdf3b) for changes.

TouwaStar commented:
Seems to work ok on my pc, will accept as soon as fog confirms

urwrstkn8mare committed:
fix: remove 'itch' in output path + create release dir if it doesn't exist

should resolve https://github.com/TouwaStar/Galaxy_Plugin_Minecraft/pull/10#discussion_r487398212

urwrstkn8mare commented:
Once everything is confirmed, I'll up ver to 1.0.2. Then the PR should be merged.

Edit: or you can do it just a reminder that it's got to be upped cause I made some changes since 1.0.1
